components/org.wso2.carbon.identity.oauth.uma.resource.endpoint/src/main/java/org/wso2/carbon/identity/oauth/uma/resource/endpoint/impl/ResourceRegistrationApiServiceImpl.java

```diff
@@ -353,7 +355,14 @@ private Response getUnauthorizedResponseObject() {
 
     private URI getResourceLocationURI(CreateResourceDTO response) throws URISyntaxException {
 
-        return new URI(RESOURCE_PATH + "/" + response.getResourceId());
+        try {
+            String locationUri = ServiceURLBuilder.create().addPath(UMAConstants.REGISTERED_RESOURCE_PATH, response
+                    .getResourceId()).build().getAbsolutePublicURL();
+            return new URI(locationUri);
+        } catch (URLBuilderException e) {
+            throw new RuntimeException("Error occurred while building the resource location URI with tenant qualified" +
```

Comment:
We can simply say "Error occurred while building the resource location URI."

Also, let's wrap the original exception